### DISCUSSION TITLE
Additional events for `--onevent` argument

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [librespot-org/librespot#638] [General]
The `onevent` API is very useful to generate additional UI ontop of `librespot`, without having to re-implement most of the CLI, however it's missing many events which could be useful.

- Device connection / disconnection
- Differing pause and stop events
- Play time (although this may not be possible with Spotify's API)

Looking through the runtime logs, there's lots of output from `librespot_connect::spirc`. Potentially these events could be channelled into the existing `PlayerEvent` flow. I feel it's better to have too many events, that the user can filter out, than too few, and I have to parse the `stdout` of the process (something I was partially tempted to do).

If someone could point me to the best location to put the call to `run_program`, I'd be happy to contribute the code myself!

**Comment by michaelherger** [comment]
Seems to be under investigation. See eg. https://github.com/ashthespy/librespot/commit/545c9eaa9732bc7fe0236289ac931185930a6d55

**Comment by RealOrangeOne** [comment]
ooh, awesome! I hope to see that merged at some point :+1:

**Comment by sashahilton00** [comment]
Thanks for getting started on this, my plan is to add events from SpIRC as well. As you said, having too many is better, as we'll eventually send them out as JSON in librespotd

**Comment by ashthespy** [comment]
Gimme a few days, will clean this up and submit a PR for a proper review :-)

I'm aiming to incorporate the events @sashahilton00 mentions at https://github.com/plietar/librespot/issues/37#issuecomment-172688099
